cluster.go

```diff
@@ -316,7 +316,25 @@ func (c *cluster) setCoordinator(n *Node) error {
 	}
 
 	// Broadcast cluster status.
-	return c.broadcaster.SendSync(c.status())
+	return c.unprotectedSendSync(c.status())
+}
+
+// unprotectedSendSync is used in place of c.broadcaster.SendSync (which is
+// Server.SendSync) because Server.SendSync needs to obtain a cluster lock to
+// get the list of nodes. TODO: the reference loop from
+// Server->cluster->broadcaster(Server) will likely continue to cause confusion
+// and should be refactored.
+func (c *cluster) unprotectedSendSync(m Message) error {
+	var eg errgroup.Group
+	for _, node := range c.nodes {
```

Comment:
Can we do this? `c.nodes` is part of the cluster, things can write to it if they have the mutex, and we're calling this without the mutex. So I think the reason Server.SendSync needed the lock is that if you access the list without the lock, it's unsafe.

I think we need to grab a copy of the node list to do this safely?

Reply:
everything that calls this should already be holding c.mu (unless I screwed something up)

Reply:
The call right above it looks to me like it just released c.mu and then called this. (Line 310 in setCoordinator).